### PR TITLE
Kerberos support for presto

### PR DESCRIPTION
**Pull request by anishanainani**
Added Kerberos support in presto.py script

- Added auth parameter to the Cursor
- By default auth is NONE but user can pass NONE/PLAIN and KERBEROS

### CODE REVIEW

**Comment by prabhu1984** [comment]
Great to see Kerberos support for presto! Can you please review and merge?

**Comment by jingw** [comment]
Sorry, haven't forgotten about this. Will take a look next week.

File: pyhive/presto.py [hunk 1]
```diff
@@ -79,7 +80,7 @@ class Cursor(common.DBAPICursor):
 
     def __init__(self, host, port='8080', username=None, catalog='hive',
                  schema='default', poll_interval=1, source='pyhive', session_props=None,
-                 protocol='http', password=None):
+                 protocol='http', password=None,auth='NONE',verify=False):
```
Comment (devinstevenson) [inline pyhive/presto.py]: wouldn't it be best to have as default `verify=True`? deactivating SSL should be explicitly done.

Comment (anishanainani) [inline pyhive/presto.py]: I put it as False because usually users would use it without SSL but I can set it to True by default if that is preferred.

**Comment by devinstevenson** [comment]
@anishanainani adding `requests_kerberos` to `tests_require` and `requires_extra` in `setup.py` should fix the test failures.

**Comment by anishanainani** [comment]
@devinstevenson I added requests_kerberos to tests_require and requires_extra in setup.py, but the tests are still failing with requests_kerberos module not found. Can you please review this? Thanks

**Comment by devinstevenson** [comment]
@anishanainani sorry, you need to add it to dev_requirements.txt file too.

**Comment by anishanainani** [comment]
@devinstevenson I added requests_kerberos in dev_requirements and did some small fixes. All the tests failures are now fixed. Can you please review this? Thanks

File: pyhive/presto.py [hunk 2]
```diff
@@ -18,6 +18,7 @@
 import requests
 from requests.auth import HTTPBasicAuth
 
+from requests_kerberos import HTTPKerberosAuth, OPTIONAL
```
Comment (jingw) [inline pyhive/presto.py]: Could you make this an optional dependency?
```
try:
    import requests_kerberos
except ImportError:
    # no kerberos support
    pass
```

File: pyhive/presto.py [hunk 3]
```diff
@@ -79,7 +80,7 @@ class Cursor(common.DBAPICursor):
 
     def __init__(self, host, port='8080', username=None, catalog='hive',
                  schema='default', poll_interval=1, source='pyhive', session_props=None,
-                 protocol='http', password=None):
+                 protocol='http', password=None, auth='NONE', verify=True):
```
Comment (jingw) [inline pyhive/presto.py]: When would someone want to pass verify=False?

File: pyhive/presto.py (hunk 2 above)
Comment (jingw) [inline pyhive/presto.py]: above comment is obsolete assuming we just go with the `requests_options` approach.

**Comment by anishanainani** [comment]
@jingw Thanks. I think thats a good idea. We should close this Pull request then and I can send a new one with requests_options.

**Comment by jingw** [comment]
I'm going to push a change to make requests options customizable.

**Pull request closed without merging**